HW0/P4/P4.py

```diff
@@ -1,28 +1,71 @@
 import numpy as np
 import matplotlib.pyplot as plt
 from mpl_toolkits.mplot3d import Axes3D
-
+from scipy.optimize import fmin
+
+	
+
+
+def kalman_filter(s_0, Sig_0, A, B, C, K, s_obs):
+	# initializations
+	Sig_k = Sig_0
+	Sig_t = Sig_k
+	s_tilde = s_0
+	s_f_pred = np.zeros((K, 6))
+	s_f_pred[0] = s_0
+	
+	for i in range(K-1):
```

Comment:
Better to use `xrange` instead of `range`, so there's no chance of memory blowup if `K` is large. `range` creates a list in memory first and then iterates over it, whereas `xrange` is a generator, which changes the counter on the fly. (`range` was changed in Python 3 to do this.)